orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -80,16 +81,41 @@ onnxruntime::MLDataType ort_scalar_type_from_aten(
 OrtValue create_ort_value(
   onnxruntime::ORTInvoker& invoker,
   const at::Scalar& scalar) {
-  // TODO: support more types
+  return create_ort_value(invoker, scalar, at::kFloat);
+}
+
+OrtValue create_ort_value(
+  onnxruntime::ORTInvoker& invoker,
+  const at::Scalar& scalar,
+  at::ScalarType type) {
   float val = scalar.toFloat();
+  uint16_t temp16{};
   OrtValue ort_val;
   CreateMLValue(
     invoker.GetCurrentExecutionProvider().GetAllocator(0, OrtMemTypeDefault),
-    ort_scalar_type_from_aten(at::kFloat),
+    ort_scalar_type_from_aten(type),
     {},
     &ort_val);
   auto* ort_tensor = ort_val.GetMutable<onnxruntime::Tensor>();
-  CopyVectorToTensor<float>(invoker, {val}, *ort_tensor);
+  switch (type) {
+    case at::ScalarType::Float:
+      CopyVectorToTensor<float>(invoker, {val}, *ort_tensor);
+      break;
+    case at::ScalarType::BFloat16:
+      ORT_IF_CONSTEXPR(onnxruntime::endian::native == onnxruntime::endian::little) {
+        std::memcpy(&temp16, reinterpret_cast<char*>(&val) + sizeof(uint16_t), sizeof(uint16_t));
```

Comment:
does at::Scalar has toBFloat16() feature?

Reply:
Yes - I have used it here.